jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -647,23 +652,23 @@ ecma_builtin_global_object_decode_uri_helper (ecma_value_t uri __attr_unused___,
       uint32_t min;
       uint32_t character;
 
-      if (decoded_byte < 0xe0)
+      if (decoded_byte < LIT_UTF8_2_BYTE_MASK)
       {
         count = 1;
-        min = 0x80;
-        character = decoded_byte & 0x1f;
+        min = LIT_UTF8_2_BYTE_CODE_POINT_MIN;
+        character = decoded_byte & LIT_UTF8_LAST_5_BITS_MASK;
       }
-      else if (decoded_byte < 0xf0)
+      else if (decoded_byte < LIT_UTF8_3_BYTE_MASK)
       {
         count = 2;
-        min = 0x800;
-        character = decoded_byte & 0x0f;
+        min = LIT_UTF8_3_BYTE_CODE_POINT_MIN;
+        character = decoded_byte & LIT_UTF8_LAST_4_BITS_MASK;
       }
       else
       {
         count = 3;
-        min = 0x1000;
-        character = decoded_byte & 0x07;
+        min = LIT_UTF8_4_BYTE_CODE_POINT_MIN;
+        character = decoded_byte & LIT_UTF8_LAST_3_BITS_MASK;
       }
```

Comment:
And also, possibly, we can change code, starting from line 755:
```
        if (decoded_byte < LIT_UTF8_2_BYTE_MASK)
        {
          count = 1;
        }
        else if (decoded_byte < LIT_UTF8_3_BYTE_MASK)
        {
          count = 2;
        }
        else
        {
          count = 3;
        }
```
with call to `lit_get_unicode_char_size_by_utf8_first_byte`.